packages/atomic/src/components/commerce/product-template-components/product-template-decorators.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  InteractiveItemContext,
+  InteractiveItemContextEvent,
+} from '@/src/decorators/item-list/interactive-item-context';
+import {
+  ItemContext,
+  ItemContextEvent,
+} from '@/src/decorators/item-list/item-context';
+import {Product} from '@coveo/headless/commerce';
+import {itemContext} from '../../common/item-list/item-context';
+
+/**
+ * A [Lit property decorator](https://lit.dev/docs/components/decorators/) to be used for product template components.
+ * This allows the Lit component to fetch the current product from its rendered parent, the `atomic-product` component.
+ *
+ * Example:
+ * ```
+ * @ProductContext() product!: Product;
```

Comment:
Our Lit decorators are usually in camelCase, not PascalCase, so this would be inconsistent.

I see that we're already exporting a `productContext` function, though.

Perhaps we could rename the decorator to something like `productItemContext`?